javascripts/2048.js

```diff
@@ -1,32 +1,347 @@
 $(document).ready(function() {
   console.log('ready!');
+
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
     if(arrow_keys.indexOf(event.which) > -1) {
       var tile = $('.tile');
       moveTile(tile, event.which);
+      addTile();
       event.preventDefault();
     }
   })
+
+  $('h6').click(function(){
+    gameStart();
+  })
 })
 
 function moveTile(tile, direction) {
-  var new_tile_value = tile.attr("data-val") * 2;
-  tile.attr("data-val", new_tile_value);
-  tile.text(new_tile_value);
+  // if (tilesCollide(direction); == true) {
+    // var new_tile_value = tile.attr("data-val") * 2;
+    // tile.attr("data-val", new_tile_value);
+    // tile.text(new_tile_value); // then also moves to next available position
+  // } else {
+    // they do not merge, they slide into the next available position
+  // }
+  var c0 = $("div[data-col='c0']");
+  var col0 = $.makeArray(c0);
+  var c1 = $("div[data-col='c1']");
+  var col1 = $.makeArray(c1);
+  var c2 = $("div[data-col='c2']");
+  var col2 = $.makeArray(c2);
+  var c3 = $("div[data-col='c3']");
+  var col3 = $.makeArray(c3);
+
+  var tile = col0.concat(col1, col2, col3);
 
   switch(direction) {
     case 38: //up
-      tile.attr("data-row","r0");
+      // tile.attr("data-row","r0"); // unless a tile is there
+      var cols = makeCols(tile);
+      moveColUp(cols);
+      collideTilesUp(cols);
+      var reTile = $('.tile');
+      var reCol = makeCols(reTile);
+      moveColUp(reCol);
       break;
```

Comment:
Having separately called move and collide functions could be causing your ghosting problem. If you made it so you checked for collision as you moved the tiles, you would also be able to skip the secondary move.